src/language-markdown/printer-markdown.js

```diff
@@ -295,13 +292,35 @@ function genericPrint(path, options, print) {
       return concat(["[^", printChildren(path, options, print), "]"]);
     case "footnoteReference":
       return concat(["[^", node.identifier, "]"]);
-    case "footnoteDefinition":
+    case "footnoteDefinition": {
+      const nextNode = path.getParentNode().children[path.getName() + 1];
       return concat([
         "[^",
         node.identifier,
         "]: ",
-        printChildren(path, options, print)
+        group(
+          concat([
+            align(
+              " ".repeat(options.tabWidth),
+              printChildren(path, options, print, {
+                processor: (childPath, index) =>
+                  index === 0
+                    ? group(
+                        concat([
+                          ifBreak(concat([line, line]), ""),
+                          childPath.call(print)
+                        ])
+                      )
+                    : childPath.call(print)
+              })
+            ),
+            nextNode && nextNode.type === "footnoteDefinition"
+              ? ifBreak(line, "")
```

Comment:
Can this be a `softline`?

Reply:
There's no actual differences between `line` and `softline` in this context since it's only printed if break, and I found it's also used in other printer this way, so I think it should be fine? 

https://github.com/prettier/prettier/blob/ea5bfb44e4faa52d9ad0b7ccf992a185de851d8d/src/language-js/printer-estree.js#L2295

Reply:
@ikatyang From what I’ve seen, `softline` is equivalent to `ifBreak(line, "")`. Is that incorrect?

Reply:
Oops, I thought you meant `ifBreak(line)` -> `ifBreak(softline)`. Fixed.